src/audio/module_adapter/module/volume/volume.c

```diff
@@ -261,86 +257,18 @@ static int32_t volume_windows_fade_ramp(struct processing_module *mod, int32_t r
 }
 #endif
 
-/**
- * \brief Ramps volume changes over time.
- * \param[in,out] mod Volume processing module handle
-
- */
-static void volume_ramp(struct processing_module *mod)
```

Comment:
moving this function to a header shouldn't be needed, the compiler should inline it automatically in .c too when building with optimisation. Could you check that keeping the function here doesn't reduce your performance gains? Keeping the function here would also help reduce the size of the patch and make changes clearer. In fact it would be even better to split this patch into separate optimisations, but at least making changes more obvious by keeping the function in place would be good

Reply:
I actually saw performance improvement from the change. It was like inline wasn't happening.

Yep, such changes are good for separate commit.

I'm actually now thinking to remove the function call pointer for ramp function and change it to if - else. Then I could inline the ramp math also.

Reply:
Sorry, no improvement from function move to volume.h. The reason was some other change, I'll return the function to volume.c.